src/main/webapp/app/exercises/programming/shared/commits-info/commits-info-group/commits-info-group.component.html

```diff
@@ -0,0 +1,35 @@
+@if (!isExpanded) {
+    <jhi-commits-info-row
+        [ngStyle]="{ display: 'contents' }"
+        [commit]="commits[0]"
+        [currentSubmissionHash]="currentSubmissionHash"
+        [previousSubmissionHash]="previousSubmissionHash"
+        [exerciseProjectKey]="exerciseProjectKey"
+        [isRepositoryView]="isRepositoryView"
+        [user]="user"
+        [isExpanded]="isExpanded"
+        [toggleExpand]="toggleExpand.bind(this)"
+        [pushNumber]="pushNumber"
+        [firstCommit]="groupIndex === groupCount - 1 && 0 === commits.length - 1"
+        [groupCommitCount]="commits.length"
+        [groupCommitIndex]="0"
+    />
+} @else {
+    @for (commit of commits; track commit; let i = $index) {
+        <jhi-commits-info-row
+            [ngStyle]="{ display: 'contents' }"
+            [commit]="commit"
+            [currentSubmissionHash]="currentSubmissionHash"
+            [previousSubmissionHash]="previousSubmissionHash"
+            [exerciseProjectKey]="exerciseProjectKey"
+            [isRepositoryView]="isRepositoryView"
+            [user]="user"
+            [isExpanded]="isExpanded"
+            [toggleExpand]="toggleExpand.bind(this)"
```

Comment:
Like coderabbit said, the change detection will call `bind` a lot. We should avoid this for performance reasons.

In `CommitsInfoRowComponent `, you could instead define an `@Output` with a suitable `EventEmitter` (`EventEmitter<void>` should be sufficient here). You can then consume the event here by doing `(insertEventEmitterNameHere)="toggleExpand()"`.